src/test/java/org/cryptomator/common/EnvironmentTest.java

```diff
@@ -22,41 +22,7 @@ public class EnvironmentTest {
 	@BeforeEach
 	public void init() {
 		env = Mockito.spy(Environment.getInstance());
-		Mockito.when(env.getHomeDir()).thenReturn(Path.of("/home/testuser"));
 	}
-
-	@Test
-	@DisplayName("cryptomator.settingsPath=~/.config/Cryptomator/settings.json:~/.Cryptomator/settings.json")
-	public void testSettingsPath() {
-		System.setProperty("cryptomator.settingsPath", "~/.config/Cryptomator/settings.json:~/.Cryptomator/settings.json");
-
-		List<Path> result = env.getSettingsPath().toList();
-		MatcherAssert.assertThat(result, Matchers.hasSize(2));
-		MatcherAssert.assertThat(result, Matchers.contains(Paths.get("/home/testuser/.config/Cryptomator/settings.json"), //
-				Paths.get("/home/testuser/.Cryptomator/settings.json")));
-	}
-
-	@Test
-	@DisplayName("cryptomator.ipcSocketPath=~/.config/Cryptomator/ipc.socket:~/.Cryptomator/ipc.socket")
-	public void testIpcSocketPath() {
-		System.setProperty("cryptomator.ipcSocketPath", "~/.config/Cryptomator/ipc.socket:~/.Cryptomator/ipc.socket");
-
-		List<Path> result = env.ipcSocketPath().toList();
-		MatcherAssert.assertThat(result, Matchers.hasSize(2));
-		MatcherAssert.assertThat(result, Matchers.contains(Paths.get("/home/testuser/.config/Cryptomator/ipc.socket"), //
-				Paths.get("/home/testuser/.Cryptomator/ipc.socket")));
-	}
-
-	@Test
-	@DisplayName("cryptomator.integrationsWin.keychainPaths=~/AppData/Roaming/Cryptomator/keychain.json")
-	public void testKeychainPath() {
-		System.setProperty("cryptomator.integrationsWin.keychainPaths", "~/AppData/Roaming/Cryptomator/keychain.json");
-
-		List<Path> result = env.getKeychainPath().toList();
-		MatcherAssert.assertThat(result, Matchers.hasSize(1));
-		MatcherAssert.assertThat(result, Matchers.contains(Paths.get("/home/testuser/AppData/Roaming/Cryptomator/keychain.json")));
-	}
-
```

Comment:
when removing all these tests, please make sure to add equivalent tests to `LazyProcessedPropertiesTest.java`

Reply:
What are these tests for? My reading is, that we want to ensure that "~" gets resolved correctly for those specific properties and that the property might contain multiple paths.

With this PR, it is not the task of `Environment.java` to resolve anything. Only parsing the content to paths. I'll readd the tests in a "slim" form again.

Reply:
What I meant to say is this: Yes, they can be removed from `EnvironmentTest.java`. But instead I would like to see them in `SubstitutingPropertiesTest.java`. Yes, you have already added tests for `process()`, but the ones I would like to keep are more "end to end":

```java
@ParameterizedTest(name = "{0}={1} -> {0}={2}")
@DisplayName("Replace @{userhome} during getProperty()")
@CsvSource(quoteCharacter = '"', textBlock = """
		cryptomator.settingsPath, "@{userhome}/.config/Cryptomator/settings.json:@{userhome}/.Cryptomator/settings.json", "/home/.config/Cryptomator/settings.json:/home/.Cryptomator/settings.json"
		cryptomator.ipcSocketPath, "@{userhome}/.config/Cryptomator/ipc.socket:@{userhome}/.Cryptomator/ipc.socket", "/home/.config/Cryptomator/ipc.socket:/home/.Cryptomator/ipc.socket"
		not.cryptomator.not.substituted, "@{userhome}/foo", "@{userhome}/foo"
		cryptomator.no.placeholder.found, "foo/bar", "foo/bar"
		// ...
		"""})
public void test(String key, String raw, String substituted) {
	var delegate = Mockito.mock(Properties.class);
	Mockito.doReturn("/home").when(delegate).getProperty("user.home");
	Mockito.doReturn(raw).when(delegate).getProperty(key);
	var inTest = new SubstitutingProperties(delegate, Map.of());

	var result = inTest.getProperty(key);

	Assertions.assertEquals(substituted, result);
}

// same for `appdir`, since this is a different "Source"
```

Reply:
Implemented in https://github.com/cryptomator/cryptomator/pull/2957/commits/ee82965ef5d7e069ba00f691508e267930850a8d